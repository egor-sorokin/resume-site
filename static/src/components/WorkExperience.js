import React, { Component } from 'react';
import { getWorkExperience } from '../utils/api';


export default class WorkExperience extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {},
      isFetching: false,
      loaded: false
    };
  }

  componentDidMount() {
    this.setState({
      isFetching: true
    });

    let workExperiencePromise = getWorkExperience();

    workExperiencePromise
      .then(res => {
        this.setState({
          data: res.data.workExperience,
          isFetching: false,
          loaded: true
        });
      })
      .catch(error => {
        console.log(`Error during fetching personal data :::: ${error}`);

        this.setState({
          isFetching: false
        });
      })
  }


  render() {
    const data = this.state.data;

    return (
      <div>
        <section>
          <p>
            {data.id}
          </p>

        </section>
      </div>
    );
  }
}
