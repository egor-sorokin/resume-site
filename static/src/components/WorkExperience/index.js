import React, { Component } from 'react';
import withFetching from '../../utils/api';
import { URL_PATH_WORK_EXPERIENCE } from '../../constants/index';


const WorkExperience = ({ data, isFetching, error }) => {
  const workExperience = data.workExperience || [];

  if (error) {
    return <p>{error.message}</p>;
  }

  if (isFetching) {
    return <p>Loading ...</p>;
  }

  return (
    <div>
      dasd
      <div>
        {workExperience.map(item =>
          <div key={item.id}>
            asd
          </div>
        )}
      </div>
    </div>
  );
};

export default withFetching(URL_PATH_WORK_EXPERIENCE, WorkExperience);
