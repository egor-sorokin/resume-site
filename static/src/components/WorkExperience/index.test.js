import React from 'react';
import WorkExperience from './index';
import { shallow, render, mount } from 'enzyme';

describe('<WorkExperience />', () => {
  const component = shallow(<WorkExperience />);

  it('renders without crashing', () => {
    shallow(<WorkExperience />);
  });

  it('renders 1 <WorkExperience /> component', () => {
    expect(component).toHaveLength(1);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders children when passed in', () => {
    const wrapper = shallow((
      <WorkExperience>
        <div className="unique" />
      </WorkExperience>
    ));
    expect(wrapper.contains(<div className="unique" />)).toEqual(true);
  });

  it('calls componentDidMount', () => {
    const spy = jest.spyOn(WorkExperience.prototype, 'componentDidMount');
    const wrapper = mount(<WorkExperience />);
    expect(spy).toHaveBeenCalledTimes(1);

    afterEach(() => {
      spy.mockClear()
    });
  });
});