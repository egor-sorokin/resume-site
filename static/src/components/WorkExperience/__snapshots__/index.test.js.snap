// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkExperience /> matches snapshot 1`] = `
<WorkExperience
  data={Object {}}
  error={null}
  isFetching={true}
/>
`;
